src/app/services/db.service.ts

```diff
@@ -51,7 +50,66 @@ export class DB {
             this.clients.set(tenantID, client);
             return client;
         }
-
         throw new Error('Invalid tenant ID');
     }
+
+    // Dans modifications: 
+    async findOrCreateChatRoom(tenantID: string, fromUserID: number, toUserID: number): Promise<number> {
+        const client: PrismaClient = this.getClient(tenantID); 
+        let room = await client.chatRoom.findFirst({
+            where: {
+                members: {
+                    every: { 
+                        userId: { in: [fromUserID, toUserID] }
+                    }
+                }
+            },
+            include: { members: true }
+        });
+        if (!room) {
+            room = await client.chatRoom.create({
+                data: {
+                    members: {
+                        createMany: {
+                            data: [
+                                { userId: fromUserID },
+                                { userId: toUserID }
+                            ],
+                            skipDuplicates: true
+                        }
+                    }
+                },
+                include: { members: true }
+            });
+        }
+        return room.id;
+    }
+
+
+    async saveMessage(tenantID: string, chatRoomId: number, fromUserId: number, toUserId: number | null, content: string): Promise<Message> {
+        const client: PrismaClient = this.getClient(tenantID); 
+        return await client.message.create({
+            data: {
+                chatRoomId,
+                fromUserId,
+                toUserId, 
+                content
+            }
+        });
+    }
+
+    async getChatHistory(tenantID: string, roomId: number): Promise<Message[]> {
+        const client: PrismaClient = this.getClient(tenantID);
+        try {
+            const messages = await client.message.findMany({
+                where: { chatRoomId: roomId },
+                orderBy: { createdAt: 'asc' }, 
+            });
+            return messages;
+        } catch (error) {
+            
+            throw error;
+        }
+    }
```

Comment:
Yeah you are right... Technically speaking these are my helper functions for my chat controller but I thought it would be better to have them in the db service since they functions are communcating with the db.

Reply:
In general, it's more important to keep things related to a particular "business function" together than it is to keep the same "technology" together. Typically if you're looking for some code you're looking for the business area, and also it means you're generally bouncing around the codebase less when trying to follow a bit of code